architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,550 @@
+
+Today, users of Radius and future adopters of Radius use many Kubernetes-specific tools in their production workflows. Most of these tools operate on Kubernetes resources exclusively - which presents a problem when trying to deploy resources defined in Bicep manifests. This design proposes the creation of a new Kubernetes controller (BicepDeployment controller) in Radius that will allow users to deploy resources defined in Bicep manifests using Kubernetes tooling.
+
+## Terms and definitions
+
+<!--
+Include any terms, definitions, or acronyms that are used in
+this design document to assist the reader. They may or may not
+be part of the user-facing experience once implemented, and can
+be specific to this design context.
+-->
+
+**CRD (Custom Resource Definition)**: A definition for a Kubernetes resource that allows users to define their own resource types.
+
+**CR (Custom Resource)**: An instance of a CRD that represents a custom resource in Kubernetes.
+
+**Bicep**: A Domain Specific Language (DSL) for defining Azure resources, or when used with Radius, Azure resources, AWS resources, and Radius resources.
+
+## Objectives
+
+<!--
+Describe goals/non-goals and user-scenario of this feature to understand
+the end-user goals.
+* If the feature shares the same objectives of the existing design, link
+  to the existing doc and section rather than repeat the same context.
+* If the feature has a scenario, UX, or other product feature design doc,
+  link it here and summarize the important parts.
+-->
+
+### Goals
+
+<!--
+Describe goals to define why we are doing this work, how we will make
+priority decisions, and how we will determine success.
+-->
+
+**Goal: Users can use Kubernetes tooling to deploy and manage resources defined in Bicep manifests**
+- With this work, users will be able to deploy resources defined in Bicep using only Kubernetes. We will essentially be providing a "translation layer" between Kubernetes resources (that Kubernetes tools can understand) and Radius resources (that Radius can understand).
+
+**Goal: Users can quickly generate a Kubernetes CR from Bicep using the Radius CLI**
+- We will provide a CLI command that generates the BicepDeployment resource from a Bicep manifest to make this feature easy to adopt.
+
+**Goal: The behavior of BicepDeployment is consistent with user expectations for a Kubernetes-enabled tool**
+- We will follow user expectations for Kubernetes controllers and resources when designing the BicepDeployment controller and resource. This includes building in support for retries, status updates, and error handling.
+
+### Non-goals
+
+<!--
+Describe non-goals to identify something that we won’t be focusing on 
+immediately. We won’t be expending any effort on these matters. If there
+will be follow-ups after this work, list them here. If there are things
+we plan to do in the future, but are out of scope of this design, list
+them here. Provide a brief explanation on why this is a non-goal.
+-->
```

Comment:
I think we'll end up with a non-goal here about the complexity of tooling we're building, or the idea of reimplementing functionality thats in the bicep compiler. Let's discuss in the context of your multi-file examples.